cryptographic_estimators/SDEstimator/SDAlgorithms/bjmm_dw.py

```diff
@@ -19,56 +19,61 @@
 from types import SimpleNamespace
 from ..sd_constants import *
+
 filterwarnings("ignore", category=RuntimeWarning)
 
 
 class BJMMdw(SDAlgorithm):
     def __init__(self, problem: SDProblem, **kwargs):
-        """
-        Construct an instance of BJMM's estimator using *d*isjoint *w*eight distributions combined with
-         MitM-nearest neighbor search. [EB22]_, [MMT11]_, [BJMM12]_.
+        """Construct an instance of BJMM's estimator using disjoint weight distributions combined with MitM-nearest neighbor search.
 
-        Expected weight distribution::
+        Expected weight distribution:
 
             +---------------------------+-------------+------------+----------+----------+----------+----------+
             |<-+ n - k - 2 l1 - 2 l2 +->|<-+ k / 2 +->|<-+ k / 2 ->|<-+ l1 +->|<-+ l1 +->|<-+ l2 +->|<-+ l2 +->|
             |   w - 2 p - 2 w1 - 2 w2   |      p      |      p     |    w1    |    w1    |    w2    |    w2    |
             +---------------------------+-------------+------------+----------+----------+----------+----------+
 
-
-        INPUT:
-
-        - ``problem`` -- syndrome decoding problem instance
-        - ``k`` -- dimension of the code
-        - ``w`` -- Hamming weight of error vector
-        - ``mem`` -- upper bound on the available memory (as log2), default unlimited
-        - ``hmap`` -- indicates if hashmap is being used (default: true)
-        - ``p_range`` -- interval in which the parameter p is searched (default: [0, 25], helps speeding up computation)
-        - ``memory_access`` -- specifies the memory access cost model (default: 0, choices: 0 - constant, 1 - logarithmic, 2 - square-root, 3 - cube-root or deploy custom function which takes as input the logarithm of the total memory usage)
-
-        EXAMPLES::
-
-            sage: from cryptographic_estimators.SDEstimator.SDAlgorithms import BJMMdw
-            sage: from cryptographic_estimators.SDEstimator import SDProblem
-            sage: BJMMdw(SDProblem(n=100,k=50,w=10))
+        Args:
+            problem (SDProblem): Syndrome decoding problem instance.
+            k (int): Dimension of the code.
+            w (int): Hamming weight of error vector.
+            mem (int, optional): Upper bound on the available memory (as log2), default unlimited.
+            hmap (bool, optional): Indicates if hashmap is being used (default: True).
+            p_range (list, optional): Interval in which the parameter p is searched (default: [0, 25], helps speeding up computation).
+            memory_access (int, optional): Specifies the memory access cost model (default: 0, choices: 0 - constant, 1 - logarithmic, 2 - square-root, 3 - cube-root or deploy custom function which takes as input the logarithm of the total memory usage).
+
+        References:
+            .. [EB22] Enguehard, Élise, and Anja Becker. "Quantum attacks on McEliece and Niederreiter cryptosystems." Designs, Codes and Cryptography 90.1 (2022): 1-23.
```

Comment:
Btw this should be [Esser, Bellini]

Reply:
I completely removed the reference section, as it didn't even exist in the original docstring.